config/placeholder_replacement.go

```diff
@@ -23,6 +24,10 @@ var placeholderRegexp = regexp.MustCompile(`\$\{(?P<placeholder_key>[^\}]*)\}`)
 // packages.FutureP4ckge_Ty-pe.name.
 var packagePlaceholderRegexp = regexp.MustCompile(`^packages(\.(?P<type>[^\.]+))?\.(?P<name>[\w:/-]+)$`)
 
+// environmentPlaceholderRegexp matches on all valid ways of specifying one of our environment placeholders
+// This is compatible with IEEE Std 1003.1-2018 (see basedefs/V1_chap08.html).
+var environmentPlaceholderRegexp = regexp.MustCompile(`^environment\.(?P<name>[\w:/-]+)$`)
```

Comment:
what was the decision ton use environment here and not {env.} I saw that the ticket indicates using environment but isn't that kind of against normal practice? 

Also have you checked that ${env.placeholder} for a local config doesn't break in this case either?

Reply:
> what was the decision ton use environment here and not {env.} I saw that the ticket indicates using environment but isn't that kind of against normal practice?

This came up in the scope and the consensus was that this gives us a lot more freedom going forward. If we just use ${env} we lock ourselves in. Ex: if there is an environment variable called `packages.ml_model.foo` what do we do?.  Or more specifically, if we decide to support other kinds placeholders in the future, we are at risk of breaking people's existing configs.

Reply:
>  Ex: if there is an environment variable called packages.ml_model.foo what do we do?

not sure what you mean by this with env vs. environment? 

Still not super sure how I follow the "locking ourselves in?"

Reply:
Oh you mean `env` vs `environment` not `${env}`

Reply:
Umm, I don't think there was a strong reason one way or the other. This is just what people were happy with.

Reply:
sorry if I am taking this above the PR, but environment is just kind of wordy and there is a standard for this which is why I am wondering where the decision is to break the standard. Especially as we support env variables in local configs, so i feel like they should be synced while we do technically support both.

Reply:
Want to talk about this offline?